src/testResultSummary/parsers/golang.ts

```diff
@@ -0,0 +1,56 @@
+import { z } from "zod"
+import {
+  SummarizedTestResults,
+  MappedTestResult,
+  TestResultsSchema,
+  handledTestResultsSchema,
+} from "../types"
+
+/**
+ * Parse a go test results file data into the expeted output while ignoring
+ * any data we do not want to be streamed to Loki
+ */
+export function parseGoTestResults(fileData: string): SummarizedTestResults {
+  const tests = parseToTestResults(fileData)
+
+  // Filter out any tests that we do not want to be streamed to Loki
+  // and map the remaining tests to the expected output
+  const handledTests: SummarizedTestResults["tests"] = tests
+    .filter(
+      (t): t is z.infer<typeof handledTestResultsSchema> =>
+        handledTestResultsSchema.safeParse(t).success,
+    )
+    .map(
+      t =>
+        ({
+          name: t.Test,
+          status: t.Action,
+          elapsed: t.Elapsed,
+        } satisfies MappedTestResult),
+    )
+
+  // The last test in the array is the summary of the test run
+  const lastTest = handledTests.at(-1)
+  if (!lastTest) {
+    throw Error("No tests found in file")
+  }
+  // Remove the last test from the array, which is the summary of the test run
+  const restOfHandledTests = handledTests.slice(0, -1)
+
+  return {
+    elapsed: lastTest.elapsed,
+    status: lastTest.status,
+    tests: restOfHandledTests,
+    testType: "go",
+  }
+}
+
+export function parseToTestResults(fileData: string) {
```

Comment:
Missing return type on function. The linter should probably be setup to catch this.

Reply:
We don't have a linter at all! It would be great to have `eslint` setup. Microsoft publishes a decent one if you want to take a look: https://www.npmjs.com/package/@rushstack/eslint-config